### ISSUE TITLE
ExponentialLR and fine tune

### ISSUE THREAD

**Issue by Alexey322** [jik876/hifi-gan#39]
Hi.
I started training the model from scratch and found that the optimizer uses a dynamic learning step. If I train the model with 2.5 million steps, then according to my calculations, the training step will drop to 3e-7. Not only is this a very low learning step that can cause floating point errors, it is also impossible to adapt other speakers at such a checkpoint, because the learning step is too small.
Does this mean that it is better to set lr_decay = 1.0?

**Comment by jik876** [comment]
Hi.
In our experiments, we observed that the learning rate decay affects stable quality improvement.
I would like to recommend adjusting the learning rate loaded from a checkpoint during transfer learning.

**Comment by Alexey322** [comment]
@jik876 
I'm not sure if this is the right thing to do, because during training with a low learning rate, the minimum is looking for smaller areas. When we change the learning rate, to a higher one from a lower one, the minimum search algorithm will search for a minimum in wider areas, respectively, everything that the model has learned at a smaller learning step will be forgotten.
It seems to me that it is more correct to train at a fixed learning rate, and then adapt the speakers on this model with a dynamic learning rate.
Maybe I'm wrong. What do you think?

**Comment by jik876** [comment]
If the purpose of training the source speaker is to make a basis for transfer learning to another speaker, then the method you mentioned may be good. If you're going to use the source speaker model as it is, refer to our experiments, using the learning rate decay leads to better quality.
The choice of learning rate in transfer learning will be affected by the difference between the source speaker and the target speaker. If the speech characteristics of the two speakers are not very different, using a small learning rate in the training will yield quite good results, and vice versa.

**Comment by LqNoob** [comment]
> If the purpose of training the source speaker is to make a basis for transfer learning to another speaker, then the method you mentioned may be good. If you're going to use the source speaker model as it is, refer to our experiments, using the learning rate decay leads to better quality.
> The choice of learning rate in transfer learning will be affected by the difference between the source speaker and the target speaker. If the speech characteristics of the two speakers are not very different, using a small learning rate in the training will yield quite good results, and vice versa.

Hi, thank you for your great project.

1. I met some question when fine tune the model using Tacotron2 feature.  Because after tuning, the sound quality has not improved, but has declined. The loss curve before and after tuning is as follows: 
before    ![image](https://user-images.githubusercontent.com/28683383/106108360-196b3b00-6183-11eb-8494-1430e48361fd.png)        after    ![image](https://user-images.githubusercontent.com/28683383/106108437-399afa00-6183-11eb-846a-c8eaa8b9c19e.png)
I don't understand why this happens.  During tuning, did you change related configuration or parameters, such as learning rate?

2. In this issues https://github.com/jik876/hifi-gan/issues/4#issuecomment-719939683. You said, "If you try a slightly different method, if you use a noisy spectrogram as the input condition and clean audio as ground truth for training, I think HiFi-GAN can synthesize clean audio from a noisy spectrogram. This is in line with the fine-tuning experiment in our paper."
There is no implementation of relevant aspects in the warehouse, so what does this sentence mean?

**Comment by Alexey322** [comment]
@LqNoob, I met the same problem as yours. I think that when we start to fine tune, the representation of the mel spectrograms is completely different from the original ones. Accordingly, the error will be the same as if you began to train a different voice. Therefore, it seems to me, you need to retrain on a small number of iterations and watch out for overfitting, because in my experiments, fine tuning caused overfitting to my surprise. The paper says that the fine tuning was done over 100k iterations, which means you probably need fewer iterations for smaller datasets. But again, it all depends on the dataset, the best way to track retraining is to look for the validation curve and listen to audio. 

Can you confirm this @jik876?
